lists2safebrowsing.py

```diff
@@ -650,7 +653,13 @@ def start_versioning(config, chunknum, shavar_prod_lists_branches):
     for branch in shavar_prod_lists_branches:
         branch_name = branch.get('name')
         ver = p_version.parse(branch_name)
+
         if isinstance(ver, p_version.Version):
+            # Delete records for a version that is no longer supported
+            if LATEST_VERSION_SUPPORTED.release[0] - ver.release[0] > int(config.get('main', 'num_supported_versions')):
```

Comment:
This method goes through all records for older versions that are not supported and tries to delete them, and if they dont exist, continues. This way, we only keep the records that are supported (last 10, or 20 etc.)

This isn't the cleanest way to do this, but the most straightforward imo without introducing any complexities in the code. 

Reasoning: it is difficult to estimate how many versions to delete without going through the records and finding the smallest version by comparing the id's. This would be a feasible option too, but would involve some extract logic to be implemented. If efficiency is a concern, we can use this method.